test/esockd_SUITE.erl

```diff
@@ -524,8 +526,82 @@ t_update_tls_options(Config) ->
     ?assertEqual(<<"Server">>, esockd_peercert:common_name(Cert1)),
     ?assertEqual(<<"Changed">>, esockd_peercert:common_name(Cert2)),
 
+    %% Changing _just_ TLS version is error-prone:
+    %% Existing TLS options might be incompatible with changed version.
+    ?assertMatch(
+       {error, #{ error := invalid_ssl_option
+                , reason := {options, _Incompatible, _}
+                }},
+        esockd:set_options({echo_tls, LPort},
+                           [{ssl_options, [{versions, ['tlsv1.3']}]}])
+    ),
+
+    %% ...And need to be explicitly reset.
+    SslOpts3 = esockd:merge_opts(SslOpts1, [{versions, ['tlsv1.3']}]),
+    ok = esockd:reset_options({echo_tls, LPort}, [{ssl_options, SslOpts3}]),
+
+    {ok, Sock3} = ssl:connect("localhost", LPort, ClientSslOpts, 1000),
+
+    ok = ssl:send(Sock1, <<"Sock1">>),
+    {ok, <<"Sock1">>} = ssl:recv(Sock1, 0, 1000),
+    ok = ssl:send(Sock2, <<"Sock2">>),
+    {ok, <<"Sock2">>} = ssl:recv(Sock2, 0, 1000),
+    ok = ssl:send(Sock3, <<"Sock3">>),
+    {ok, <<"Sock3">>} = ssl:recv(Sock3, 0, 1000),
+
     ok = esockd:close(echo_tls, LPort).
 
+t_update_dtls_options(Config) ->
```

Comment:
What property does this test verify? How do I know if the test is wrong or the code?